cpp/src/arrow/engine/substrait/ext_test.cc

```diff
@@ -158,10 +152,10 @@ TEST_P(ExtensionIdRegistryTest, ReregisterFunctions) {
   auto provider = std::get<0>(GetParam());
   auto registry = provider->get();
 
-  for (util::string_view name : kFunctionNames) {
-    auto id = Id{kArrowExtTypesUri, name};
-    ASSERT_RAISES(Invalid, registry->CanRegisterFunction(id, name.to_string()));
-    ASSERT_RAISES(Invalid, registry->RegisterFunction(id, name.to_string()));
+  for (Id function_id : kFunctionIds) {
```

Comment:
Should this be `const Id& function_id` instead?

Reply:
TBH, I don't really know.  Small things we pass by value (e.g. `int`).  A `string_view` is a pair of pointers and generally considered small enough that it is [better to pass by value](https://quuxplusone.github.io/blog/2021/11/09/pass-string-view-by-value/).

An `Id` is a pair of `string_view` objects.  Does that mean it is small enough?  I'm not sure what the exact rules are here but the existing usage seems to be treating it as a small value object and so I have tried to continue to do so.

Perhaps @bkietz for a second opinion / further guidelines.

Reply:
For anything trivial(ly copyable, destructible, ...) by-reference and by-value should be interchangeable. Using it by value here is more readable so do that. The compiler will have no difficulty removing the trivial copy